plugin/kubernetes/ns_test.go

```diff
@@ -96,8 +98,23 @@ func (APIConnTest) EpIndexReverse(ip string) []*object.Endpoints {
 					},
 				},
 			},
-			Name:      "dns6-service",
+			Name:      "dns6-service-aaaaa",
```

Comment:
please make this meaningful or add a completely new test

Reply:
Thanks. I've refactored, added comments and made the test data more compact and (i think) easier to read.  The test I had modified wasn't testing the right thing, and fixing it exposed a bug in nsAddrs, which is not fixed.